### PR TITLE
feat(423): Allow descriptions in yaml file

### PR DESCRIPTION
**Pull request by joelseq**
- Add a schema for description
- Add a test data file (config.job.description.yaml)
- Add a test to validate description

### CODE REVIEW

File: config/job.js [hunk 1]
```diff
@@ -64,6 +64,7 @@ const SCHEMA_STEPS = Joi.array().items(SCHEMA_STEP).min(1);
 const SCHEMA_IMAGE = Joi.string();
 const SCHEMA_SETTINGS = Joi.object().optional();
 const SCHEMA_TEMPLATE = Joi.string().regex(Regex.FULL_TEMPLATE_NAME);
+const SCHEMA_DESCRIPTION = Joi.string().optional();
```
Comment (FenrirUnbound) [inline config/job.js]:
```
Joi.string().max(100).optional()
```

Comment (FenrirUnbound) [inline config/job.js]:
```
Joi.string().max(100).optional()
```

Comment (ian-fox) [inline config/job.js]: Maximum length?

File: config/job.js [hunk 2]
```diff
@@ -73,7 +74,8 @@ const SCHEMA_JOB = Joi.object()
         secrets: SCHEMA_SECRETS,
         settings: SCHEMA_SETTINGS,
         steps: SCHEMA_STEPS,
-        template: SCHEMA_TEMPLATE
+        template: SCHEMA_TEMPLATE,
+        description: SCHEMA_DESCRIPTION
```
Comment (tkyi) [inline config/job.js]: alphabetical order please

File: config/job.js [hunk 3]
```diff
@@ -64,9 +64,11 @@ const SCHEMA_STEPS = Joi.array().items(SCHEMA_STEP).min(1);
 const SCHEMA_IMAGE = Joi.string();
 const SCHEMA_SETTINGS = Joi.object().optional();
 const SCHEMA_TEMPLATE = Joi.string().regex(Regex.FULL_TEMPLATE_NAME);
+const SCHEMA_DESCRIPTION = Joi.string().max(100).optional();
```
Comment (r3rastogi) [inline config/job.js]: alphabetical order here too please

Comment (joelseq) [inline config/job.js]: Right now a lot of them are not alphabetically sorted so should I just move `SCHEMA_DESCRIPTION` above `SCHEMA_IMAGE` or sort them all?

Comment (FenrirUnbound) [inline config/job.js]: I feel like if you order the block of variables that `SCHEMA_DESCRIPTION` was added to, that would be sufficient

Comment (r3rastogi) [inline config/job.js]: ^ what Darren said

**Comment by joelseq** [comment]
@r3rastogi @FenrirUnbound I sorted the block but I left `SCHEMA_JOB` at the bottom. I added a space for separation.

**Pull request merged**